src/test/scala/com/microsoft/hyperspace/index/E2EHyperspaceRulesTests.scala

```diff
@@ -419,49 +419,104 @@ class E2EHyperspaceRulesTests extends HyperspaceSuite with SQLHelper {
+        val dataFileNames = dataPath
+          .getFileSystem(new Configuration)
+          .globStatus(dataPath)
+          .map(_.getPath)
+
+        assert(dataFileNames.nonEmpty)
+        val fileToDelete = dataFileNames.head
+        FileUtils.delete(fileToDelete)
+
+        def query2(): DataFrame =
+          spark.read.parquet(location).filter("c3 == 'facebook'").select("c3", "c1")
+
+        // Verify index is not used.
+        spark.enableHyperspace()
+        val planRootPaths = getAllRootPaths(query2().queryExecution.optimizedPlan)
+        spark.disableHyperspace()
+        assert(planRootPaths.equals(Seq(PathUtils.makeAbsolute(location))))
+
+        // Refresh the index to remove deleted source data file records from index.
+        hyperspace.refreshIndex(indexConfig.indexName)
+
+        // Verify index usage on latest version of index (v=1) after refresh.
+        verifyIndexUsage(query2, getIndexFilesPath(indexConfig.indexName, 1))
+      }
+    }
+  }
 
-      // Save a copy of source data files.
-      val location = testDir + "ixRefreshTest"
-      val dataPath = new Path(location, "*parquet")
-      val dataColumns = Seq("c1", "c2", "c3", "c4", "c5")
-      SampleData.save(spark, location, dataColumns)
-
-      // Create index on original source data files.
-      val df = spark.read.parquet(location)
-      val indexConfig = IndexConfig("filterIndex", Seq("c3"), Seq("c1"))
-      hyperspace.createIndex(df, indexConfig)
-
-      // Verify index usage for index version (v=0).
-      def query1(): DataFrame =
-        spark.read.parquet(location).filter("c3 == 'facebook'").select("c3", "c1")
-
-      verifyIndexUsage(query1, getIndexFilesPath(indexConfig.indexName))
-
-      // Delete some source data file.
-      val dataFileNames = dataPath
-        .getFileSystem(new Configuration)
-        .globStatus(dataPath)
-        .map(_.getPath)
-
-      assert(dataFileNames.nonEmpty)
-      val fileToDelete = dataFileNames.head
-      FileUtils.delete(fileToDelete)
-
-      def query2(): DataFrame =
-        spark.read.parquet(location).filter("c3 == 'facebook'").select("c3", "c1")
-
-      // Verify index is not used.
-      spark.enableHyperspace()
-      val planRootPaths = getAllRootPaths(query2().queryExecution.optimizedPlan)
-      spark.disableHyperspace()
-      assert(planRootPaths.equals(Seq(PathUtils.makeAbsolute(location))))
-
-      // Refresh the index to remove deleted source data file records from index.
-      hyperspace.refreshIndex(indexConfig.indexName)
-
-      // Verify index usage on latest version of index (v=1) after refresh.
-      verifyIndexUsage(query2, getIndexFilesPath(indexConfig.indexName, 1))
-      FileUtils.delete(dataPath)
+  test("Validate index usage for enforce delete on read after some source data file is deleted.") {
+    withSQLConf(
+      IndexConstants.INDEX_LINEAGE_ENABLED -> "true",
+      IndexConstants.REFRESH_LOGENTRY_ACTION_ENABLED -> "true") {
```

Comment:
As I read this code, `REFRESH_LOGENTRY_ACTION_ENABLED` does not seem very informative. Was there any discussion around this?

Reply:
Yes, this is based on some other comment in this PR. @imback82 suggested to capture both excluded and appended source data files and change the action name to RefreshLogEntryAction. Here is the comment: https://github.com/microsoft/hyperspace/pull/170#discussion_r497039610

Reply:
Looks like the comment is around the action name, not the config?

Reply:
Yes, not the config. To me `REFRESH_LOGENTRY_ACTION` is not really clear (but it could just be me).

Reply:
btw, isn't this config temporary until we have a public API?

Reply:
@imback82 So far we had meaningful mapping between configs and corresponding action they trigger. Do you suggest using a config name s.a. `REFRESH_FOR_DELETE_ON_READ` to trigger an Action that updates Index metadata and adds both `excluded` and `appended` files?

Reply:
How about REFRESH_LOG_ENTRY_ONLY or something along that line without ACTION? I think the confusion from @rapoth may be the fact that action was included, which is internal?

Reply:
> Yes, not the config. To me `REFRESH_LOGENTRY_ACTION` is not really clear (but it could just be me).

Oh wait, are you saying the action name is not clear? Any other suggestion then?

Reply:
I was thinking something like: `RefreshSourceFilesAction` or `RefreshSourceFilesListAction` or `RefreshSourceMetadataAction` (assuming I understood this right). I just feel `RefreshLogEntryAction` is too generic and not conveying what that thing is doing. Does any of this make sense?

Reply:
what about this?
for action: `RefreshSourceContentAction`
for config: `REFRESH_SOURCE_CONTENT`

Reply:
@pirz Let me think. In the mean time, can you update the PR appropriately please? (assume these are the renames we will do)

Reply:
@rapoth - Sure, can we just finalize what we wanna rename `RefreshDeleteActionBase` (this is the "parent" class)?
 It would be great to come to a consensus as each rename could affect many comments and code pieces and I would rather do it once.